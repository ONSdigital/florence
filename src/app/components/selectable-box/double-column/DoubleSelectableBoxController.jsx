import React from "react";
import PropTypes from "prop-types";
import DoubleSelectableBoxItem from "./DoubleSelectableBoxItem";

const propTypes = {
    headings: PropTypes.array.isRequired,
    search: PropTypes.string,
    items: PropTypes.arrayOf(
        PropTypes.shape({
            id: PropTypes.string.isRequired,
            selectableBox: PropTypes.shape({
                firstColumn: PropTypes.string.isRequired,
                secondColumn: PropTypes.string.isRequired,
            }),
        })
    ).isRequired,
    activeItemID: PropTypes.string,
    handleItemClick: PropTypes.func.isRequired,
    isUpdating: PropTypes.bool,
};

const DoubleSelectableBoxController = props => {
    const renderList = () => {
        return (
            <ul id="selectable-box" className="selectable-box__list">
                {props.items.map((item, index) => {
                    return (
                        <DoubleSelectableBoxItem
                            key={index}
                            {...item}
                            isSelected={props.activeItemID && item.id === props.activeItemID}
                            handleClick={() => props.handleItemClick(item.id)}
                        />
                    );
                })}
            </ul>
        );
    };
    const renderMessage = () => {
        if (props.search) {
            return (
                <div className="selectable-box__message">
                    <p>
                        <strong>Cannot find collection</strong>
                    </p>
                    <p>Improve your results by:</p>
                    <ul>
                        <li>double-checking your spelling</li>
                        <li>searching for something less specific</li>
                    </ul>
                </div>
            );
        } else {
            return (
                <div className="selectable-box__message">
                    <p>No items to display</p>
                </div>
            );
        }
    };

    return (
        <div className="selectable-box">
            <div className="grid">
                <h2 className="selectable-box__heading grid__col-6">{props.headings[0]}</h2>
                <h2 className="selectable-box__heading grid__col-6 grid__cell">
                    {props.headings[1]}
                    {props.isUpdating && <span className="selectable-box__status pull-right loader" />}
                </h2>
            </div>
            {props.items.length > 0 ? renderList() : renderMessage()}
        </div>
    );
};

DoubleSelectableBoxController.propTypes = propTypes;

export default DoubleSelectableBoxController;
