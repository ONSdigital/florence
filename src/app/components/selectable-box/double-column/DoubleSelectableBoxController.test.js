import React from "react";
import DoubleSelectableBoxController from "./DoubleSelectableBoxController";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { items } from "../../../../tests/mockData";

const props = {
    headings: ["Foo", "Bar"],
    items,
    activeItemID: items[1].id,
    isUpdating: false,
    search: "baz",
    handleItemClick: jest.fn(),
};

describe("DoubleSelectableBoxController", () => {
    it("matches the snapshot", () => {
        const component = renderer.create(<DoubleSelectableBoxController {...props} />);
        expect(component.toJSON()).toMatchSnapshot();
    });

    describe("when there are not items", () => {
        it("displays message about search result if search props is passed", () => {
            const emptyProps = {
                ...props,
                items: [],
            };
            const component = shallow(<DoubleSelectableBoxController {...emptyProps} />);
            expect(component.text()).toEqual(
                "FooBarCannot find collectionImprove your results by:double-checking your spellingsearching for something less specific"
            );
        });
        it("displays generic message", () => {
            const emptyProps = {
                ...props,
                search: "",
                items: [],
            };
            const component = shallow(<DoubleSelectableBoxController {...emptyProps} />);
            expect(component.text()).toEqual("FooBarNo items to display");
        });
    });
});
