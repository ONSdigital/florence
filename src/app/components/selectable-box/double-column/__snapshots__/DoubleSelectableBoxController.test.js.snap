// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoubleSelectableBoxController matches the snapshot 1`] = `
<div
  className="selectable-box"
>
  <div
    className="grid"
  >
    <h2
      className="selectable-box__heading grid__col-6"
    >
      Foo
    </h2>
    <h2
      className="selectable-box__heading grid__col-6 grid__cell"
    >
      Bar
    </h2>
  </div>
  <ul
    className="selectable-box__list"
    id="selectable-box"
  >
    <li
      className="selectable-box__item   "
      id="1"
      onClick={[Function]}
    >
      <div
        className="grid"
      >
        <div
          className="grid__col-6"
        >
          Collection 1
          
        </div>
        <div
          className="grid__col-6"
        >
          [test]
        </div>
      </div>
    </li>
    <li
      className="selectable-box__item  selected  "
      id="2"
      onClick={[Function]}
    >
      <div
        className="grid"
      >
        <div
          className="grid__col-6"
        >
          Collection 2
          
        </div>
        <div
          className="grid__col-6"
        >
          Friday
        </div>
      </div>
    </li>
    <li
      className="selectable-box__item   "
      id="3"
      onClick={[Function]}
    >
      <div
        className="grid"
      >
        <div
          className="grid__col-6"
        >
          Collection 3
          
        </div>
        <div
          className="grid__col-6"
        >
          [manual collection]
        </div>
      </div>
    </li>
  </ul>
</div>
`;
